lib/fathead/pep/parse.py

```diff
@@ -0,0 +1,233 @@
+        - set_structure() - sets the structure for the file
+    """
+    collection = set()
+    document = None
+    pep = None
+    title = None
+    header = None
+    body = None
+    url = None
+
+    def __init__(self, links):
+        for link in links:
+            self.get_pep_contents(link)
+            self.set_pep_number()
+            self.set_pep_title()
+            self.set_pep_body()
+            self.set_structure()
+            print("Parsed " + self.pep)
+
+    def get_pep_contents(self, link):
+        """
+        Gets the contents from the HTML file.
+        """
+        self.url = __PYBASE__ + link['href']
+        html = request.urlopen(self.url)
+        self.set_pep_document(html.read().decode("utf-8"))
+
+    def set_pep_document(self, doc):
+        """
+        Makes up the HTML document.
+        """
+        self.document = BeautifulSoup(doc, "html.parser")
+
+    def set_pep_title(self):
+        """
+        Gets the title for the PEP
+        """
+        # peptitle = self.document.find("h1", { "class": "page-title" }).get_text()
+        # self.title = re.sub("--", "-", peptitle)
+        self.title = "PEP " + self.pep
+
+    def set_pep_number(self):
+        """
+        Sets the PEP number. For example, if the PEP is identified as PEP 0008,
+        it will be stored as 8. If the PEP is identified as PEP 0143 it will be
+        stored as 143.
+        """
+        pep = re.findall(r'\d+', self.url)[0]
+        self.pep = pep.lstrip("0")
+
+    def set_pep_body(self):
+        """
+        Parses the raw document, cleans the text
+        """
+        try:
+            html = self.document.find("div", { "id": [
+                    "abstract",
+                    "introduction",
+                    "rationale",
+                    "motivation",
+                    "what-is-a-pep",
+                    "overview",
+                    "improving-python-zip-application-support",
+                    "scope",
+                    "abstract-and-rationale",
+                    "rationale-and-goals",
+                    "specification"
+                ]
+            })
+
+            if html is None: raise AttributeError
+
+            html = FHTEMPLATE.format(information=str(html))
+            html = html.replace("<pre class=\"literal-block\">", "<pre><code>")
+            html = html.replace("</pre>", "</code></pre>")
+            html = html.replace("<tt class=\"docutils literal\">", "<code class=\"inline\">")
+            html = html.replace("</tt>", "</code>")
+            html = re.sub(re.compile("<h1>((.|\n)+?)</h1>"), "", html)
+            html = re.sub(re.compile("<a (.+?)>"), "", html)
+            html = re.sub(re.compile("</a>"), "", html)
+            html = re.sub(re.compile("\[\d+\]"), "", html) # removes the vancouver type referencing
+            html = re.sub(re.compile("\n"), " ", html)
+            html = html.rstrip().replace("\t", "\\t")
```

Comment:
You can also use 2 spaces if you prefer, as long as there are no `\\t` in the output.txt 👍